### PR TITLE
bugfix(#28): fixed `make prod` bug

### LINKED ISSUE #28: running "make production": no such service: postgrest
I'm trying to run CHT Sync against a "production" CHT instance which is really just a an instance of [CHT Docker Helper](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/app-developer/#cht-docker-helper-for-4x). This has an external to CHT Sync, URL, port,  login and password.  Here's the entries from my `.env` file:

```
COUCHDB_USER=medic
COUCHDB_PASSWORD=password
COUCHDB_DB=medic-users-meta
COUCHDB_HOST=192-168-0-37.local-ip.medicmobile.org
COUCHDB_PORT=10447
COUCHDB_SECURE=true
```

When I run `make prod` I get the following output:

```
➜  cht-sync git:(main) make prod
rm -rf ./data/json_docs
docker-compose -f docker-compose.couchdb.yml -f docker-compose.postgres.yml -f docker-compose.yml \
	down -v --remove-orphans
docker-compose build 
[+] Building 0.0s (0/2)                                                                                                                                                                                            
[+] Building 0.2s (2/3)                                                                                                                                                                                            
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.0s
 => => transferring dockerfile: 212B                                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                               0.0s
[+] Building 0.4s (7/8)                                                                                                                                                                                            
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.0s
 => => transferring dockerfile: 212B                                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                               0.0s
 => [internal] load metadata for docker.io/library/logstash:7.14.1                                                                                                                                            0.4s
 => [1/3] FROM docker.io/library/logstash:7.14.1@sha256:8f1969b7dfe7fffc13f55d72b6001804e45c1f655b8169bc84a6dfeda481c2b9                                                                                      0.0s
 => [internal] load build context                                                                                                                                                                             0.0s
 => => transferring context: 106B                                                                                                                                                                             0.0s
[+] Building 0.4s (8/8) FINISHED                                                                                                                                                                                   
 => [internal] load build definition from Dockerfile                                                                                                                                                          0.0s
 => => transferring dockerfile: 212B                                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                                             0.0s
 => => transferring context: 2B                                                                                                                                                                               0.0s
 => [internal] load metadata for docker.io/library/logstash:7.14.1                                                                                                                                            0.4s
 => [1/3] FROM docker.io/library/logstash:7.14.1@sha256:8f1969b7dfe7fffc13f55d72b6001804e45c1f655b8169bc84a6dfeda481c2b9                                                                                      0.0s
 => [internal] load build context                                                                                                                                                                             0.0s
 => => transferring context: 106B                                                                                                                                                                             0.0s
 => CACHED [2/3] COPY ./pipeline /usr/share/logstash/pipeline                                                                                                                                                 0.0s
 => CACHED [3/3] COPY ./entrypoint.sh /usr/local/bin/entrypoint.sh                                                                                                                                            0.0s
 => exporting to image                                                                                                                                                                                        0.0s
 => => exporting layers                                                                                                                                                                                       0.0s
 => => writing image sha256:999e357ca35ffbb05bd7838c566ba69ef47644ab92c029c9e4f20a9d4597e239                                                                                                                  0.0s
 => => naming to docker.io/medicmobile/cht-logstash-couchdb:latest                                                                                                                                            0.0s
docker-compose up logstash postgrest dbt
no such service: postgrest
make: *** [Makefile:19: prod] Error 1
```

**Steps to reproduce**:

0. start local CHT instance in docker helper 
1. clone this repo locally: `git clone https://github.com/medic/cht-sync`
2. change directory: `cd cht-sync`
3. copy env file: `cp env.template .env`
4. edit `.env` to have couchdb credentials from zeroeth step
5. start local instance: `make prod`

**expected**: cht sync starts
**actual**: cht sync fails to start and shows error

### PR DESCRIPTION
**Pull request by samuelimoisili**
Added a docker compose file to fix `make prod` postgrest startup issue.

Closes #28

### CODE REVIEW

**Comment by andrablaj** [comment]
I managed to reproduce @mrjones-plip's [bug](https://github.com/medic/cht-sync/issues/28) locally on the `main` branch.

If I run `make prod `on this PR's branch (after running CHT locally with the docker helper), I get the following error in the `dbt` container:

> Unable to connect! could not translate host name "postgres" to address: Name or service not known

**Comment by samuelimoisili** [comment]
@andrablaj You'll to setup a PostgreSQL database and update the following environment variables:

```env
DBT_POSTGRES_HOST=
DBT_POSTGRES_USER=
DBT_POSTGRES_PASSWORD=
POSTGRES_DB=
POSTGRES_TABLE=
DBT_POSTGRES_SCHEMA=
POSTGRES_SCHEMA=
```

**Comment by mrjones-plip** [comment]
@andrablaj  - I'm waaaay behind on my mail/slack/todo queue - removing myself as a reviewer to unblock.

**Comment by andrablaj** [comment]
@samuelimoisili, following up here with some items from our 1:1 discussion:

- I didn't see any mention in the documentation about installing PostgreSQL as a prerequisite to run `cht-sync`. Can you please add that step in the `README`?
- You mentioned having a docker-compose file for setting up PostgreSQL locally. Can you please share it in the repo with instructions on how to use it and how it reflects in the` .env` file variables?  

Thanks!

**Review by andrablaj — CHANGES_REQUESTED** [review]
🎆 Awesome, I managed to run `make prod` successfully locally, with the latest CHT run via docker helper. Great job! 

Nevertheless, the documentation needs updates as I needed to perform extra steps to have this running correctly. 

1. Content of `.env`:

```shell
# project wide: optional
COMPOSE_PROJECT_NAME=pipeline

# postgrest and pogresql: required environment variables for 'gamma', prod and 'local'
POSTGRES_USER=root
POSTGRES_PASSWORD=supercoolpassword
POSTGRES_DB=data
POSTGRES_TABLE=couchdb
POSTGRES_SCHEMA=v1

# dbt: required environment variables for 'gamma', 'prod' and 'local'
DBT_POSTGRES_USER=dbt_user
DBT_POSTGRES_PASSWORD=supercoolpassword
DBT_POSTGRES_SCHEMA=dbt
DBT_POSTGRES_HOST=yourIP
CHT_PIPELINE_BRANCH_URL=https://github.com/medic/cht-pipeline.git#main

# couchdb and logstash: required environment variables for 'gamma', 'prod' and 'local'
COUCHDB_USER=medic
COUCHDB_PASSWORD=password
COUCHDB_DB=medic-users-meta
COUCHDB_HOST=yourCHThost
COUCHDB_PORT=yourCHTport
COUCHDB_SECURE=false

# superset: required environment variables for 'gamma', 'prod' and 'local'
SUPERSET_PASSWORD=password
SUPERSET_ADMIN_EMAIL=user@superset.com
```
2. Run `make prod`
3. Run `docker-compose -f docker-compose.postgres.yml -f docker-compose.yml up postgres`.

I think it's worth mentioning that the user can skip step 3 if they have a local instance of postgreSQL, and update `.env` accordingly.

File: env.template [hunk 1]
```diff
@@ -12,6 +12,7 @@ POSTGRES_SCHEMA=v1
 DBT_POSTGRES_USER=dbt_user
 DBT_POSTGRES_PASSWORD=supercoolpassword
 DBT_POSTGRES_SCHEMA=dbt
+DBT_POSTGRES_HOST=postgres # Your postgres instance IPA or endpoint in "prod".
```
Comment (andrablaj) [inline env.template:15]:
```suggestion
DBT_POSTGRES_HOST=postgres # Your postgres instance IP or endpoint in "prod".
```

**Review by andrablaj — APPROVED** [review]
:shipit:

**Pull request merged**